blockchain/sync_manager/src/sync.rs

```diff
@@ -78,32 +74,25 @@ impl<'a> Syncer<'a> {
         Ok(())
     }
     /// Returns message root CID from bls and secp message contained in the param Block
-    fn compute_msg_data(&self, block: &Block) -> Result<Cid, CidError> {
+    fn compute_msg_data(&self, block: &Block) -> Result<Cid, Error> {
+        // retrieve bls and secp messages
         let bls_msgs = block.bls_msgs();
         let secp_msgs = block.secp_msgs();
-
-        let mut bls_cids = Vec::new();
-        for bls_msg in bls_msgs {
-            bls_cids.push(bls_msg.cid());
-        }
-
-        let mut secp_cids = Vec::new();
-        for secp_msg in secp_msgs {
-            secp_cids.push(secp_msg.cid());
-        }
-
-        let mut amt = AMT::new(&self.chain_store.db);
+        // collect bls and secp cids
+        let bls_cids = bls_msgs.iter().map(|x| x.cid()).collect::<Vec<_>>();
+        let secp_cids = secp_msgs.iter().map(|x| x.cid()).collect::<Vec<_>>();
```

Comment:
You're actually creating an AMT of results, should just use the cids_from_messages helper function you have in this file. How did this function go unused when it was previously used?

Should look something like:
```
        let bls_cids = cids_from_messages(block.bls_msgs())?;
        let secp_cids = cids_from_messages(block.secp_msgs())?;
```

Reply:
you removed the function with: https://github.com/ChainSafe/forest/pull/196/commits/9155c84212b43d3c43bf098dc50dd73649c130f7, but you probably want it as it will make your life easier